### PR TITLE
RecoveryServices.Backup Bugfixes

### PR DESCRIPTION
**Pull request by dragonfly91**
## Description

1. Fixed bug - Bug 1526979: Get-AzureRmRecoveryServicesBackupItem should do case insensitive comparison for container name filter
2. Fixed bug - AzureVmItem should have property displaying ServiceClientModel.AzureIaaSVMProtectedItem.LastBackupTime
3. Fixed bug - Make vault credential cmdlet test compulsory for Check-In
4. Improvement - made tests self-sustaining. 
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/199/

File: src/ResourceManager/RecoveryServices/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Current Release
+* Fixed bug - Make vault credential cmdlet test compulsory for Check-In.
```
Comment (cormacpayne) [inline src/ResourceManager/RecoveryServices/ChangeLog.md]: @dragonfly91 if the only changes in this module were test changes, then you can go ahead and revert the changes made to this change log. There shouldn't be a need to ship another iteration of `RecoveryServices` if there were no changes impacting customers.

**Comment by dragonfly91** [comment]
Hi @cormacpayne I have undone my changes made to the RecoveryServices ChangeLog.md file. Please merge the PR!

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/222/

**Comment by maddieclayton** [comment]
@twitchax looks like the travis build isn't running for this PR - is there an easy way to retrigger it?

**Comment by maddieclayton** [comment]
Closing and reopening to trigger the Travis job

**Comment by twitchax** [comment]
@maddieclayton, sorry, my bad for the delay.  This PR needed a forward integrate from preview.  I merged, and the CI should be kicked off now.

**Comment by maddieclayton** [comment]
No changes were made to common files that would break the netcore build, so I will just force merge this.
